src/extension.ts

```diff
@@ -50,8 +50,10 @@ export function activate(context: ExtensionContext) {
       updateSavedInfo()
     }),
     // Listen for text editor focus
-    window.onDidChangeWindowState(() => {
-      updateSavedInfo()
+    window.onDidChangeWindowState(e => {
+      if (e.focused) {
```

Comment:
Previously, this was being triggered whether the window was focused or not. This introduced a race condition where the window being closed would end up being entered as the saved bit of info.